packages/metal-component/src/Component.js

```diff
@@ -650,6 +676,16 @@ class Component extends EventEmitter {
 	validatorEventsFn_(val) {
 		return !isDefAndNotNull(val) || isObject(val);
 	}
+
+	/**
+	 * Lifecycle. Fires before the component has been attached to the DOM.
+	 */
+	willAttach() {}
+
+	/**
+	 * Lifecycle. Fires before component is detached from the DOM.
+	 */
+	willDetach() {}
```

Comment:
Should we also have an empty placeholder for `willUpdate`? I see we are actually checking down the line to see if `component.willUpdate` exists, but it feels a bit inconsistent with how we are treating every other lifecycle method.

The most similar case seems to be `rendered`, which is actually always used from the `component` instance `informRendered` wrapper method...